tests/common/fixtures.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import os
+
+import pytest
+
+
+ELASTICSEARCH_HOST = os.environ.get("ELASTICSEARCH_HOST", "http://localhost:9200")
+ELASTICSEARCH_INDEX = "hypothesis-test"
+
+
+@pytest.fixture
+def search_client():
+    """A :py:class:`h.search.client.Client` for the test search index."""
+    return _search_client()
+
+
+@pytest.fixture(scope="session", autouse=True)
+def init_elasticsearch(request):
+    """Initialize the test Elasticsearch index once per test session."""
+    from h.search import init
+
+    client = _search_client()
+
+    def maybe_delete_index():
+        """Delete the test index if it exists."""
+        if client.conn.indices.exists(index=ELASTICSEARCH_INDEX):
+            client.conn.indices.delete(index=ELASTICSEARCH_INDEX)
+
+    # Delete the test search index at the end of the test run.
+    request.addfinalizer(maybe_delete_index)
+
+    # Delete the test search index at the start of the run, just in case it
+    # was somehow left behind by a previous test run.
+    maybe_delete_index()
+
+    # Initialize the test search index.
+    init(client)
+
+
+@pytest.fixture(autouse=True)
+def delete_all_elasticsearch_documents(request):
+    """Delete everything from the test search index after each test."""
+    client = _search_client()
+
+    def delete_everything():
+        client.conn.delete_by_query(index=client.index, body={"query": {"match_all": {}}})
```

Comment:
Just a note for me with the new version: if we are still using delete by query i need to install the plugin in the docker container.